File: templates/Enterprise-Template/src/csharp/EnterpriseBotTemplate/Bot Framework/Dialogs/Escalate/EscalateResponses.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using $safeprojectname$.Dialogs.Escalate.Resources;
using $safeprojectname$.Extensions;
using Microsoft.Bot.Builder.TemplateManager;
using Microsoft.Bot.Schema;
using System.Collections.Generic;

namespace $safeprojectname$
{
    public class EscalateResponses : TemplateManagerWithVoice
    {
        public const string SendPhone = "sendPhone";

        private LanguageTemplateDictionary _responseTemplates = new LanguageTemplateDictionary
        {
            ["default"] = new TemplateIdMap
            {
                { SendPhone, SendEscalateCard },
            },
            ["en"] = new TemplateIdMap { },
            ["fr"] = new TemplateIdMap { },
        };

        public EscalateResponses()
        {
            Register(new DictionaryRenderer(_responseTemplates));
        }

        public static IMessageActivity SendEscalateCard(ITurnContext turnContext, dynamic data)
        {
            var response = turnContext.Activity.CreateReply();

            response.Speak = EscalateStrings.PHONE_INFO;
            response.Attachments = new List<Attachment>
            {
                new HeroCard()
                {
                    Text = EscalateStrings.PHONE_INFO,
                    Buttons = new List<CardAction>()
                {
                    new CardAction(type: ActionTypes.OpenUrl, title: "Call now", value: "tel:8001235555"),
                    new CardAction(type: ActionTypes.OpenUrl, title: "Open Teams", value: "msteams://")
                },
                }.ToAttachment()
            };

            return response;
        }
    }
}

File: templates/Enterprise-Template/src/typescript/src/dialogs/escalate/escalateResponses.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { ActionTypes, Activity, CardFactory, TurnContext } from "botbuilder";
import { ActivityExtensions } from "../../extensions/activityExtensions";
import { DictionaryRenderer, LanguageTemplateDictionary, TemplateFunction } from "../templateManager/dictionaryRenderer";
import { TemplateManager } from "../templateManager/templateManager";
import * as i18n from "i18n";

export class EscalateResponses extends TemplateManager {
    
    // Fields
    public static ResponseIds = {
        SendPhoneMessage:  "sendPhoneMessage",
    }
    private static readonly _responseTemplates: LanguageTemplateDictionary = new Map([
        ["default", new Map([
            [EscalateResponses.ResponseIds.SendPhoneMessage, EscalateResponses.fromResources("escalate.phoneInfo")]
        ])]
    ]);

    constructor() {
        super();
        this.register(new DictionaryRenderer(EscalateResponses._responseTemplates));
    }

    public static async buildEscalateCard(turnContext: TurnContext, data: any): Promise<Activity> {
        
        const response = ActivityExtensions.createReply(turnContext.activity);
        const text = i18n.__("escalate.phoneInfo");
        response.attachments = [CardFactory.heroCard(
            text,
            undefined,
            [
                {
                    title: 'Call now',
                    type: ActionTypes.OpenUrl,
                    value: 'tel:18005551234',  
                },
                {
                    title: 'Open Teams',
                    type: ActionTypes.OpenUrl,
                    value: 'msteams://',  
                },
            ],
        )];

        return Promise.resolve(response);
    }

    private static fromResources(name: string): TemplateFunction {
        return (context: TurnContext, data: any) => Promise.resolve(i18n.__(name));
    }
}
